From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Fri, 21 Oct 2022 14:16:37 +0200
Subject: [PATCH] Build: Patch Serenity root directory into libtool

PHP's libtool does not have sysroot support; this is the minimum change
to get PHP to build.
---
 build/ltmain.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/build/ltmain.sh b/build/ltmain.sh
index 2f1c8c9dc80f095832c58aeb30ca65fc27e8a64b..cce4f76267a211775ec97eb2b006df536cd4cc7b 100755
--- a/build/ltmain.sh
+++ b/build/ltmain.sh
@@ -2353,6 +2353,9 @@ EOF
 	  continue
 	  ;;
 	esac # case $deplib
+	if [ "${lib::1}" == '=' ]; then
+		lib="${SERENITY_INSTALL_ROOT}${lib:1}"
+	fi
 	if test "$found" = yes || test -f "$lib"; then :
 	else
 	  $echo "$modename: cannot find the library \`$lib' or unhandled argument \`$deplib'" 1>&2
@@ -2469,6 +2472,7 @@ EOF
 	laname=`$echo "X$lib" | $Xsed -e 's%^.*/%%'`
 
 	# Find the relevant object directory and library name.
+	libdir="${SERENITY_INSTALL_ROOT}${libdir}"
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
 	    $echo "$modename: warning: library \`$lib' was moved." 1>&2
